src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -38,6 +38,24 @@ public boolean contains(Person toCheck) {
         return internalList.stream().anyMatch(toCheck::isSamePerson);
     }
 
+    /**
+     * Returns the Student that has the same student id as {@code id}
+     */
+    public Student getStudentWithId(NusNetId id) {
+        requireNonNull(id);
+        for (int i = 0; i < internalList.size(); i++) {
+            Person p = internalList.get(i);
+            if (!(p instanceof Student)) {
+                continue;
+            }
+            Student s = (Student) p;
+            if (s.getStudentId().equals(id)) {
+                return s;
+            }
+        }
+        throw new PersonNotFoundException();
```

Comment:
maybe we can throw a StudentNotFoundException instead? it's inside the exceptions package of person!